lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -101,6 +101,49 @@ EscapeAnalysis::findRecursivePointerKind(SILType Ty,
   return NoPointer;
 }
 
+// Return the PointerKind that summarized an object's stored properties.
+//
+// If a class only holds fields of non-pointer types, then it is guaranteed not
+// to point to any other objects.
+EscapeAnalysis::PointerKind
+EscapeAnalysis::findObjectKind(SILType Ty, const SILFunction &F) const {
+  if (Ty.isAddress())
+    return AnyPointer;
+
+  auto *classDecl = Ty.getClassOrBoundGenericClass();
+  if (!classDecl)
+    return AnyPointer;
+
+  auto &M = F.getModule();
+  auto expansion = F.getTypeExpansionContext();
+
+  // Start with the most precise pointer kind
+  PointerKind objectKind = NoPointer;
+  auto meetAggregateKind = [&](PointerKind otherKind) {
+    if (otherKind > objectKind)
+      objectKind = otherKind;
+  };
+  for (Type classTy = Ty.getASTType(); classTy;
+       classTy = classDecl->getSuperclass()) {
+    classDecl = classTy->getClassOrBoundGenericClass();
+    assert(classDecl && "superclass must be a class");
+
+    // Return AnyPointer unless we have guaranteed visibility into all class and
+    // superclass properties.
+    if (classDecl->isResilient())
```

Comment:
You probably want `classDecl->isResilient(F.getModule(), F.getResilienceExpansion())` to make the analysis less conservative in the case where the class is resilient, but `F` is defined in the same module and is not inlinable.

Reply:
oh... I was specifically worried about inlinable code. I guess `F.getResilienceExpansion` already takes that into consideration